zk/src/org/zkoss/zk/ui/util/Clients.java

```diff
@@ -613,4 +617,40 @@ public static final void syncErrorbox() {
 	public static final void syncErrorbox(Component component) {
 		response(new AuSyncErrorbox(component));
 	}
+
+	/**
+	 * Focus the component (HtmlBasedComponent)
+	 * @param component target component
+	 * @since 9.5.0
+	 */
+	public static final void focus(Component component) {
+		if (component instanceof HtmlBasedComponent)
+			((HtmlBasedComponent) component).focus();
+	}
+
+	/**
+	 * Focus the first matching component (HtmlBasedComponent) by using selector
+	 * @param selector the selector string
+	 * @since 9.5.0
+	 */
+	public static final void focus(String selector) {
+		List<Component> comps = Selectors.find(Executions.getCurrent().getDesktop().getFirstPage(), selector);
+		if (comps.size() > 0)
+			focus(comps.get(0));
+		else
+			throw new UiException("Component not found by the selector: " + selector);
+	}
+
+	/**
+	 * Scrolls the ancestor elements to make the specified element visible by using selector
+	 * @param selector the selector string
+	 * @since 9.5.0
+	 */
+	public static final void scrollIntoView(String selector) {
```

Comment:
1. add all String version of Component methods
2. extract selector to component